source/core/shape.cc

```diff
@@ -1321,34 +1321,35 @@ RBFShape<dim>::gradient(const Point<dim> &evaluation_point,
   Tensor<1, dim> dr_dx_derivative{};
   Tensor<1, dim> gradient{};
 
-  if (iterable_nodes.size() > 0)
+  if (iterable_nodes.size() == 0)
     {
-      for (const size_t &node_id : iterable_nodes)
-        {
-          // Calculation of the dr/dx
-          relative_position =
-            (evaluation_point - rotated_nodes_positions[node_id]);
-          distance            = (relative_position).norm();
-          normalized_distance = distance / support_radii[node_id];
-          if (distance > 0.0)
-            dr_dx_derivative = relative_position / distance;
-          else
-            for (int d = 0; d < dim; d++)
-              // Can be proved by taking the limit (definition of a derivative)
-              dr_dx_derivative[d] = 1.0;
-          // Calculation of the dr_norm/dr
-          drnorm_dr_derivative = 1.0 / support_radii[node_id];
-          // Calculation of the d(basis)/dr
-          dbasis_drnorm_derivative =
-            evaluate_basis_function_derivative(basis_functions[node_id],
-                                               normalized_distance);
-          // Sum
-          gradient += dbasis_drnorm_derivative * drnorm_dr_derivative *
-                      dr_dx_derivative * weights[node_id];
-        }
+      throw std::logic_error(
+        "Every location inside the bounding box should be covered by the radius "
+        "of at least one RBF node. It this isn't the case, an error has been "
+        "introduced in the code. ");
+    }
+  for (const size_t &node_id : iterable_nodes)
+    {
+      // Calculation of the dr/dx
+      relative_position = (evaluation_point - rotated_nodes_positions[node_id]);
+      distance          = (relative_position).norm();
+      normalized_distance = distance / support_radii[node_id];
+      if (distance > 1e-16)
+        dr_dx_derivative = relative_position / distance;
+      else
+        for (int d = 0; d < dim; d++)
+          // Can be proved by taking the limit (definition of a derivative)
```

Comment:
```suggestion
          // Can be proven by taking the limit (definition of a derivative)
```
Thisdoes not explain what is being done here. That's what the comment should give as an information.